diff --git a/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle b/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
index 4ec4514..0077909 100644
--- a/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
+++ b/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
@@ -80,6 +80,10 @@ ext.useExpoPublishing = {
     apply plugin: 'maven-publish'
   }
 
+  if (components.findByName("release") == null) {
+    return
+  }
+
   project.android {
     publishing {
       singleVariant("release") {
